### PR TITLE
gpu: Add detection of destroyed descriptor resources

### PR DESCRIPTION
**Pull request by jeremyg-lunarg**
Add a bit array tracking which descriptor resources are valid so that it can be checked by the GPUAV instrumentation if a descriptor is used by a shader.

### CODE REVIEW

File: layers/gpu_validation/gpu_validation.cpp [hunk 1]
```diff
@@ -1122,23 +1194,35 @@ void GpuAssisted::PreCallRecordCreateShadersEXT(VkDevice device, uint32_t create
...
+                 << debug_record[kInstBindlessBoundsOutDescIndex] << " used to index descriptor array of length " << debug_record[kInstBindlessBoundsOutDescBound] << ". "
+                 << " param6=" <<  debug_record[kInstBindlessBoundsOutUnused];
             vuid_msg = "UNASSIGNED-Descriptor index out of bounds";
         } break;
         case kInstErrorBindlessUninit: {
+            const uint32_t param5 = debug_record[kInstBindlessUninitOutUnused];
+            const uint32_t param6 = debug_record[kInstBindlessUninitOutUnused2];
             strm << "(set = " << debug_record[kInstBindlessUninitOutDescSet] << ", binding = " << debug_record[kInstBindlessUninitOutBinding] << ") Descriptor index "
-                 << debug_record[kInstBindlessUninitOutDescIndex] << " is uninitialized.";
+                 << debug_record[kInstBindlessUninitOutDescIndex] << " is uninitialized." << " param5=" << param5 << " param6=" << param6;
```
Comment (Tony-LunarG) [inline layers/gpu_validation/gpu_validation.cpp:1216]: Are param5 and param6 supposed to be part of the error message?  Are they for debugging?

File: layers/sync/sync_validation.cpp [hunk 2]
```diff
@@ -1706,7 +1706,10 @@ struct PipelineBarrierOp {
             layout_transition = false;
         }
     }
-    PipelineBarrierOp(const PipelineBarrierOp &) = default;
+
+    PipelineBarrierOp(const PipelineBarrierOp &rhs)
```
Comment (Tony-LunarG) [inline layers/sync/sync_validation.cpp:1710]: Is this needed for GPU-AV?

**Review by Tony-LunarG — COMMENTED** [review]
Great work and nice cleanup along the way.  Just a couple nits.

File: layers/gpu_shaders/gpu_shaders_constants.h [hunk 3]
```diff
@@ -19,6 +19,18 @@
...
 
+#ifdef __cplusplus
+namespace gpuav_glsl {
+using uint = unsigned int;
+
+// Upper bound for maxUpdateAfterBindDescriptorsInAllPools. This value needs to
+// be small enough to allow for a table in memory, but some devices set it to 2^32.
+// This value only matters for host code, but it is defined here so it can be used
+// in unit tests.
+const uint kDebugInputBindlessMaxDescriptors = 1024u*1024u;
```
[the remark this answers is not in the document]
Comment (jeremyg-lunarg) [reply layers/gpu_shaders/gpu_shaders_constants.h:30]: It is for the reason that the comment I added above it explains. :)

File: layers/gpu_validation/gpu_validation.cpp (hunk 1 above)
Comment (jeremyg-lunarg) [reply layers/gpu_validation/gpu_validation.cpp:1216]: yeah that was for debugging and I'll take it out.

File: layers/sync/sync_validation.cpp (hunk 2 above)
Comment (jeremyg-lunarg) [reply layers/sync/sync_validation.cpp:1710]: this is needed for android builds and it is in a separate commit in the pr

File: layers/gpu_validation/gpu_descriptor_set.cpp [hunk 4]
```diff
@@ -0,0 +1,371 @@
...
+    // though the application has not requested it.
+    if (gv_dev->api_version >= VK_API_VERSION_1_2) {
+        layout_.device_addr = DispatchGetBufferDeviceAddress(gv_dev->device, &buffer_device_address_info);
+    } else {
+        layout_.device_addr = DispatchGetBufferDeviceAddressKHR(gv_dev->device, &buffer_device_address_info);
+    }
+    assert(layout_.device_addr != 0);
+
+    vmaUnmapMemory(gv_dev->vmaAllocator, layout_.allocation);
+
```
[the remark this answers is not in the document]
Comment (jeremyg-lunarg) [reply layers/gpu_validation/gpu_descriptor_set.cpp:146]: it wasn't cached, but I made it so and added the flush.

**Review by Tony-LunarG — APPROVED** [review]

**Pull request merged**